File: Data Structures & Algorithms/max-consecutive-ones/submission-1.java
class Solution {
    public int findMaxConsecutiveOnes(int[] nums) {
        int maxOnes = 0;
        int ones = 0;
        for (int i = 0; i < nums.length; i++){
            if (nums[i] == 1) {
                ones++;
            }
            else if (nums[i] != 1){
                ones = 0;
            }
            if (ones > maxOnes){
                maxOnes = ones;
            }
        }
        return maxOnes;
    }

}

File: Data Structures & Algorithms/combination-target-sum/submission-1.py
class Solution:
    def combinationSum(self, nums: List[int], target: int) -> List[List[int]]:
        results = []
        def backtrack(startIndex, currSum, currCombination) -> List[List[int]]:
            if currSum == target:
                results.append(currCombination.copy())
                currCombination = []
                currSum = 0
                return
            if currSum > target:
                return
            if startIndex >= len(nums):
                return
            currSum += nums[startIndex]
            currCombination.append(nums[startIndex])
            backtrack(startIndex, currSum, currCombination)
            currCombination.pop()
            currSum -= nums[startIndex]
            backtrack(startIndex + 1, currSum, currCombination)



        backtrack(0, 0, [])
        return results
